source/unified-test-format/unified-test-format.rst

```diff
@@ -195,6 +195,19 @@ error if an attempt is made to store an entity with a name that already exists
 in the map and MUST raise an error if an entity is not found for a name or is
 found but has an unexpected type.
 
+Test runners MUST provide a mechanism to retrieve entities from the entity
+map prior to the clearing of the entity map, as discussed in
+`Executing a Test`_. There MUST be a way to retrieve an entity by its name
+(for example, to support retrieving the iteration count stored by the
+``storeIterationsAsEntity`` option). Test runners MAY permit or require
+constraining the entity to be retrieved with its expected type.
+
+Test runners MAY restrict access to driver objects (e.g. MongoClient,
+ChangeStream) and only allow access to BSON types (see:
+`Supported Entity Types`_). This restriction may be necessary if the
+test runner needs to ensure driver objects in its entity map are properly
```

Comment:
> This restriction may be necessary if the test runner needs to ensure driver objects in its entity map are properly freed/destroyed between tests.

This sentence seems to indicate that it should be possible to access to BSON types stored in the entity map after all the tests in the `tests` array have been run by the test runner? But, this section also states:

> To ensure each test is executed in isolation, test runners MUST NOT share entity maps between tests.

These two statements seem to be contradictory. I think access to the BSON data in entity map would be accessible while running a `test` (https://github.com/mongodb/specifications/blob/master/source/unified-test-format/unified-test-format.rst#test) but not necessarily after depending on the implementation. In Python, for instance, we would facilitate any access to BSON members of the entity map prior to performing post-test cleanup. 

I suggest removing this sentence entirely.

Reply:
It is possible for a test runner to expose the entity map after the test completes, for example in garbage-collected languages. In languages with explicit memory management this may be impossible. For this reason the time when entity map can be accessed is explicitly defined (it's right before it would be cleared at the end of each test).

A particular language implementation can copy the entity map at the required time and access the copy indefinitely from that point onward.

Reply:
Since this is specified as a "MAY" clause, I am fine with keeping it as-is.